### PR TITLE
Enable firesim to use conda

### PR DESCRIPTION
**Pull request by timsnyder-siv**
**Related PRs or Issues**:
firesim/firesim#986

**Type of change**:
- [ ] Bug fix
- [X] New feature
- [ ] Other enhancement

**Impact**:
- [ ] RTL change
- [ ] Software change (RISC-V software)
- [ ] Build system change
- [X] Other 

**Contributor Checklist**:
- [X] Did you set `main` as the base branch?
- [X] Did you delete any extraneous prints/debugging code?
- [X] (If applicable) Did you add documentation for the feature?
- [X] (If applicable) Did you add a test demonstrating the PR?

- [X] (If applicable) Did you mark the PR as "Please Backport"?

**Release Notes**

Slightly modify build_toolchains.sh to enable it to build using the conda tools.

### CODE REVIEW

File: scripts/build-toolchains.sh
Comment (abejgonzalez) [inline scripts/build-toolchains.sh:157]: Can you put a comment here indicating the bug with boost.

**Review by abejgonzalez — COMMENTED** [review]
This seems fine to me but CI is failing since you aren't on a more recent version of Chipyard. Could you rebase this branch on `main` to properly run the CI.

File: scripts/build-toolchains.sh
Comment (timsnyder-siv) [reply scripts/build-toolchains.sh:157]: Sure.  I'll do that and rebase to main

Comment (timsnyder-siv) [reply scripts/build-toolchains.sh:159]: done.

**Review by davidbiancolin — COMMENTED** [review]
These changes seem reasonable to me.

**Comment by timsnyder-siv** [comment]
@abejgonzalez can I get an approval to merge this?

**Review by abejgonzalez — APPROVED** [review]
LGTM

**Review by abejgonzalez — CHANGES_REQUESTED** [review]
Can you re-build the toolchain in CI just to confirm that things work? Once that is done I'm fine with merging. 

https://github.com/ucb-bar/chipyard/blob/46d902943e7ff88fa2e620af43c0ab894f0730dd/.github/workflows/chipyard-run-tests.yml#L10

**Review by abejgonzalez — APPROVED** [review]
LGTM

**Comment by timsnyder-siv** [comment]
@abejgonzalez , can you click the checkmark now that I've invalidated the tool cache? I'd like to get this merged while there aren't any conflicts.

**Comment by abejgonzalez** [comment]
I LGTM'ed. I'm happy to merge this if you want

**Comment by timsnyder-siv** [comment]
@abejgonzalez please merge.  I didn't realize I wouldn't be able to once it was approved and passing.

**Pull request merged**
